### PR TITLE
Implement the routine load process of Kafka on Backend

### PR DESCRIPTION
**Pull request by morningman**
1. Extract the common part of stream load utilities from stream_load.cpp, such as StreamLoadContext, StreamLoadExecutor, move it to `runtime/stream_load/`

2. Implement the routine load task executor, and can consuming data via kafka client. All implementations are in `runtime/routine_load/`

### CODE REVIEW

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    // create TopicPartitions
+    std::vector<RdKafka::TopicPartition*> topic_partitions;
+    for (auto& entry : _ctx->kafka_info->begin_offset) {
+        RdKafka::TopicPartition* tp1 = RdKafka::TopicPartition::create(
+                _ctx->kafka_info->topic, entry.first, entry.second);
+        topic_partitions.push_back(tp1);
+    }
+
+    // delete TopicPartition finally
+    auto tp_deleter = [] (const std::vector<RdKafka::TopicPartition*>& vec) {
```
Comment (imay) [inline be/src/runtime/routine_load/data_consumer.cpp]:
```suggestion
    auto tp_deleter = [&topic_partitions] () {
```

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        std::for_each(vec.begin(), vec.end(),
+                [](RdKafka::TopicPartition* tp1) { delete tp1; });
+    };
+    DeferOp delete_tp(std::bind<void>(tp_deleter, topic_partitions));
```
Comment (imay) [inline be/src/runtime/routine_load/data_consumer.cpp]: no need to use `std::bind`

File: be/src/runtime/routine_load/routine_load_task_executor.cpp [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            break;
+        default:
+            std::stringstream ss;
+            ss << "unknown routine load task type: " << ctx->load_type;
+            err_handler(ctx, Status::CANCELLED, ss.str());
+            cb(ctx);
+            return;
+    }
+
+#define HANDLE_ERROR(stmt, err_msg) \
```
Comment (imay) [inline be/src/runtime/routine_load/routine_load_task_executor.cpp]: don't use macro here, you can create another function do following work, and this function call that function, and checks its return and handler error

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    }
+
+    RdKafka::Conf *conf = RdKafka::Conf::create(RdKafka::Conf::CONF_GLOBAL);
+    
+    // conf has to be deleted finally
+    auto conf_deleter = [] (RdKafka::Conf *conf) { delete conf; };
+    DeferOp delete_conf(std::bind<void>(conf_deleter, conf));
+
+    std::string errstr;
+#define SET_KAFKA_CONF(conf_key, conf_val) \
```
Comment (imay) [inline be/src/runtime/routine_load/data_consumer.cpp]: use lambda instead

File: fe/src/main/java/org/apache/doris/load/routineload/RLTaskTxnCommitAttachment.java [hunk 5]
```diff
@@ -29,20 +29,20 @@
 // "numOfTotalData": "", "taskId": "", "jobId": ""}
 public class RLTaskTxnCommitAttachment extends TxnCommitAttachment {
 
-    public enum RoutineLoadType {
+    public enum LoadSourceType {
```
Comment (EmmyMiao87) [inline fe/src/main/java/org/apache/doris/load/routineload/RLTaskTxnCommitAttachment.java]: There is a `LoadDataSourceType`enum in package `routineload`.  Using 'LoadDataSourceType' is better

**Pull request merged**